src/plugins/ccode/testmod_ccode.cpp

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include "ccode.hpp"
+
+#include <kdbmodule.h>
+#include <kdbprivate.h>
+
+#include <tests.hpp>
+
+using std::string;
+
+using CppKeySet = kdb::KeySet;
+using CppKey = kdb::Key;
+
+using ckdb::elektraModulesClose;
+using ckdb::elektraModulesInit;
+using ckdb::elektraPluginClose;
+using ckdb::elektraPluginOpen;
+using ckdb::keyNew;
+using ckdb::ksDel;
+
+using ckdb::Plugin;
+
+CppKeySet defaultConfig ()
+{
+	CppKeySet config{ 20,
+			  keyNew ("user/chars", KEY_END),
+			  keyNew ("user/chars/0A", KEY_VALUE, "6E", KEY_END), // new line -> n
+			  keyNew ("user/chars/20", KEY_VALUE, "77", KEY_END), // space -> w
+			  keyNew ("user/chars/23", KEY_VALUE, "72", KEY_END), // # -> r
+			  keyNew ("user/chars/5C", KEY_VALUE, "62", KEY_END), // \\ (backslash) -> b
+			  keyNew ("user/chars/3D", KEY_VALUE, "65", KEY_END), // = -> e
+			  keyNew ("user/chars/3B", KEY_VALUE, "73", KEY_END), // ; -> s
+			  KS_END };
+	return config;
+}
+
+CppKeySet percentConfig ()
+{
+	CppKeySet config{ 20,
+			  keyNew ("user/chars", KEY_END),
+			  keyNew ("user/chars/0A", KEY_VALUE, "6E", KEY_END), // new line -> n
+			  keyNew ("user/chars/20", KEY_VALUE, "77", KEY_END), // space -> w
+			  keyNew ("user/chars/23", KEY_VALUE, "72", KEY_END), // # -> r
+			  keyNew ("user/chars/5C", KEY_VALUE, "62", KEY_END), // \\ (backslash) -> b
+			  keyNew ("user/chars/3D", KEY_VALUE, "65", KEY_END), // = -> e
+			  keyNew ("user/chars/3B", KEY_VALUE, "73", KEY_END), // ; -> s
+			  keyNew ("user/escape", KEY_VALUE, "25", KEY_END),   // use % as escape character
+			  KS_END };
+	return config;
+}
+
+void testRoundTrip (string const decodedString, string const encodedString = "", CppKeySet config = defaultConfig ())
+#ifdef __llvm__
+	__attribute__ ((annotate ("oclint:suppress[empty if statement]"), annotate ("oclint:suppress[high cyclomatic complexity]"),
```

Comment:
Does the code below really emits all these problems? Maybe we should create Macros with these attributes?

Reply:
Does the code below really emits all these problems?

Yes. The main problems seem to be the macros `succeed_if_same` and `exit_if_fail`. Every time we use them OCLint prints the following two warnings:

- “too few branches in switch statement”, and
- “empty if statement”

.

> Maybe we should create Macros with these attributes?

Nah, I think it is okay as it is. These annotations should not look pretty. Ideally we should fix the underlying problems, instead of using annotations to disable the warnings.